src/orion/core/io/cmdline_parser.py

```diff
@@ -0,0 +1,247 @@
+                formatted.append(item)
+            else:
+                formatted.append(item.format(**configuration))
+
+        return formatted
+
+    def parse(self, commandline):
+        """Parse the `commandline` argument.
+
+        This function will go through the list of strings and create a dictionary
+        mapping every argument to its value(s). It will also form a template to recreate such a
+        list. If `commandline` is empty or the `parse` function has already been called, it
+        will return the dictionary of the already-parsed arguments.
+
+        Parameters
+        ----------
+        commandline : list
+            List of string representing the commmandline arguments.
+
+        Returns
+        -------
+        OrderedDict
+            Dictionary holding the values of every argument. The keys are the name of the arguments.
+
+        Raises
+        ------
+        ValueError
+            This exception is raised then the parser detects a duplicate argument.
+
+        Notes
+        -----
+        Each argument is stored inside an `OrderedDict` to preserve its position inside the
+        commandline string. Inside that dictionary, the keys are created following these rules:
+
+        -If the argument is a positional one, its key will be `'_pos_x'` where `x` is its index
+        inside the list.
+        -Otherwise, the key will be equal to the name of the argument minus the prefixed dashes.
+
+        The values are stored in the following ways:
+
+        -If the argument is a positional one, its value is simply its value inside the commandline.
+        -If the argument is a named, boolean argument, its value is simply `True`.
+        -If the argument is a named, single-valued argument, its value is stored as-is.
+        -If the argument is a named, multi-valued argument, its value is a list containing each
+        value following it until the next named argument.
+
+        Positional arguments following a named argument are not currently supported.
+        Aggregation of single characters arguments is not supported yet. Ex: `-xzvf`
+
+        Examples
+        --------
+
+        Positional and named arguments:
+
+        >>> parser = CmdlineParser()
+        >>> parser.parse('python 1 --arg value'.split(' '))
+        OrderedDict([('_pos_0', 'python'), ('_pos_1', '1'), ('arg', 'value')])
+
+        Named boolean argument:
+
+        >>> parser.parse('python --boolean'.split(' '))
+        OrderedDict([('_pos_0', 'python'), ('boolean', True)])
+
+        Named multi-valued argument:
+
+        >>> parser.parse('python --args value1 value2'.split(' '))
+        OrderedDict([('_pos_0', 'python'), ('args', ['value1', 'value2'])])
+
+        Named argument defined with `=`:
+
+        >>> parser.parse('python --arg=value'.split(' '))
+        OrderedDict([('_pos_0', 'python'), ('arg', 'value')])
+
+        """
+        if not commandline or self._already_parsed:
```

Comment:
That's likely to be confusing. I remember having a use-case for that in Kleio because of the branching but I don't see any use case for Oríon. I'd say if already_parse raise RuntimeError('Already parsed').